tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/config/ITmfConfiguration.java

```diff
@@ -43,5 +43,5 @@ public interface ITmfConfiguration {
      * @return optional informational parameters to return. Can be used to show
      *         more details to users of the configuration instance.
      */
-    Map<String, Object> getParameters();
+    Map<String, String> getParameters();
```

Comment:
Some design discussion is needed for this because it changes the parameter to a map<String, String> instead of map<String, Object>. This allows to serialize the object from the base class and there is no need for each extension to implement their own serialization. However, this removes the possibility to have more complex values, e.g. arrays or built-in types like requested_items. Besides it's not clear when using requested_items as parameters how to serialize, because these values are not REST-ful and changes between executions. Hence, other data needs to be serialized instead of the current requested_items values (maybe use tree path).

Reply:
I kinda think that gson would serialize like
"item: { a: [1,2,3,4,5,6]} " No?

Reply:
If we change value to be a string, then FE can't use an array as value. The FE will have to provide a String with requested items encoded somehow. That's different than to other endpoints where we use requested_items. requested_items was just one example. Other TSP specified query-parameters have similar issues.

The other issue is that that class cannot blindly serialize the parameters as they come from the FE because some of them are not REST-Ful, example requested_items. Not clear how to handle that. For that we would need a custom serializer.